packages/atomic/src/components/search/generatedAnswer/atomic-citation/atomic-citation.tsx

```diff
@@ -0,0 +1,162 @@
+import {GeneratedAnswerCitation, InteractiveCitation} from '@coveo/headless';
+import {
+  createPopper,
+  preventOverflow,
+  Instance as PopperInstance,
+} from '@popperjs/core';
+import {Component, h, State, Prop, Element, Watch} from '@stencil/core';
+import {Heading} from '../../../common/heading';
+import {LinkWithResultAnalytics} from '../../../common/result-link/result-link';
+
+/**
+ * @internal
+ */
+@Component({
+  tag: 'atomic-citation',
+  styleUrl: 'atomic-citation.pcss',
+})
+export class AtomicCitation {
+  @Element() public host!: HTMLElement;
+
+  @Prop() citation!: GeneratedAnswerCitation;
+  @Prop() index!: number;
+  @Prop() sendHoverEndEvent!: (citationHoverTimeMs: number) => void;
+  @Prop() interactiveCitation!: InteractiveCitation;
+
+  @State() public isOpen = false;
+
+  private citationRef!: HTMLElement;
+  private popupRef!: HTMLElement;
+  private popperInstance?: PopperInstance;
+  private stopPropagation?: boolean;
+
+  private hoverStart?: number;
+  private hoverAnalyticsTimeout = 1000;
+
+  private hoverTimeout?: ReturnType<typeof setTimeout>;
+  private hoverDebounceTimeoutMs = 200;
+
+  @Watch('isOpen')
+  sendHoverAnalytics() {
+    if (this.isOpen) {
+      this.hoverStart = new Date().getTime();
+      return;
+    } else {
+      if (!this.hoverStart) {
+        return;
+      }
+      const difference = new Date().getTime() - this.hoverStart;
+      if (difference > this.hoverAnalyticsTimeout) {
+        this.sendHoverEndEvent(difference);
+      }
+      this.hoverStart = undefined;
+    }
```

Comment:
redundant else block
```suggestion
    if (!this.hoverStart) {
      return;
    }
    const difference = new Date().getTime() - this.hoverStart;
    if (difference > this.hoverAnalyticsTimeout) {
      this.sendHoverEndEvent(difference);
    }
    this.hoverStart = undefined;
```